test/lib/sdkdriver/src/testdriver/browser/base.ts

```diff
@@ -0,0 +1,89 @@
+import { EventEmitter } from 'events';
+import { Server as HTTPServer } from 'http';
+import TestDriver from '../';
+import { createWebServer } from '../../util/index';
+
+interface BrowserDriverOptions {
+  browser: 'chrome' | 'firefox';
+  host: string;
+  params?: any;
+}
+
+/**
+ * Browser process driver.
+ * @class
+ * @classdesc A {@link BrowserDriver} manages a browser process where the
+ *   cross-browser test is running. It also manages a web server which serves
+ *   the code that sets up the SDK in the browser.
+ */
+abstract class BrowserDriver extends EventEmitter implements TestDriver {
+  private readonly _webServer: HTTPServer;
+  protected readonly _options: BrowserDriverOptions;
+  abstract startBrowser(url: string): Promise<void>;
+  abstract stopBrowser(): void;
+
+  /**
+   * Constructor.
+   * @param {BrowserDriverOptions} options
+   */
+  constructor(options: BrowserDriverOptions) {
+    super();
+    this._options = options;
+    this._webServer = createWebServer();
+  }
+
+  /**
+   * Get the web server.
+   * @returns {HTTPServer}
+   */
+  get webServer(): HTTPServer {
+    return this._webServer;
+  }
```

Comment:
Does this need to be public?

Reply:
@markandrus We need to pass this to the `WSServerTransport` after it is started.